README.rst

```diff
@@ -209,6 +209,32 @@ The underlying reason is, that services like *Read the Docs* sometimes change
 the working directory for good reasons and using the installed metadata
 prevents using needless volatile data there.
 
+Usage from Docker
+-----------------
+
+By default, docker will not copy the ``.git``  folder into your container.
+Therefore, builds with version inference might fail.
+Consequently, you can use the following snipped to infer the version from the host os without copying the entire ``.git`` folder to your Dockerfile.
+
+.. code:: dockerfile
+
+    RUN --mount=source=.git,target=.git,type=bind \
+        pip install --no-cache-dir -e .
+
+However, this build step introduces a dependency to the state of your local
+.git folder the build cache.
+To optimize build caching, one can use ``SETUPTOOLS_SCM_PRETEND_VERSION=1`` in
```

Comment:
for safety/santiy across other packages, this should use `SETUPTOOLS_SCM_PRETEND_VERSION_FOR_$PKG_NAME`

in addition a recommendation to have a optional build arg to set that version to something realistic would be very important

Reply:
Thank you. This is an other option I did not consider. My approach was to use the PRETEND_VERSION feature only in the particular layer, thus the version number will not be used, because it is overwritten by the next instruction. In my setup the additional line speeds up the build form 1 minute to 5 seconds. I will elaborate on that.

Reply:
I inspected this a bit further. The fist variant
```dockerfile
FROM python
COPY pyproject.toml
RUN --mount=source=.git,target=.git,type=bind \
    pip install --no-cache-dir -e .
```
 generates slightly smaller images

![Screenshot from 2021-11-16 10-02-26](https://user-images.githubusercontent.com/2777736/141955337-239fef5c-ee61-4099-bada-1a6632a14012.png)

However, the second 
```dockerfile
FROM python
COPY pyproject.toml
RUN SETUPTOOLS_SCM_PRETEND_VERSION=1 pip install -e .[test]
RUN --mount=source=.git,target=.git,type=bind pip install -e .
```
is much faster for repetitive builds.
![Screenshot from 2021-11-16 09-48-37](https://user-images.githubusercontent.com/2777736/141955443-be908b2d-c716-490d-a29e-b78106e64e3e.png)

I adjusted the documentation slightly. As I am totally new to this project, it would be highly appreciated that you perform further edits on this PR to ensure a coherent README.